pkg/parcacol/querier.go

```diff
@@ -600,9 +605,7 @@ func (q *Querier) queryRangeNonDelta(ctx context.Context, filterExpr logicalplan
 
 		// TODO: This still queries way too much data from the underlying database.
 		// This needs to be moved to FrostDB to not even query all of this data in the first place.
-		// With a scrape interval of 10s and a query range of 1d we'd query 8640 samples and at most return 960.
-		// Even worse for a week, we'd query 60480 samples and only return 1000.
-		tsBucket := ts / 1000 / int64(step.Seconds())
+		tsBucket := ts / int64(step.Seconds())
```

Comment:
This worked as intended. We want to divide by 1000 as we store the timestamps as milliseconds. 
Then for each step, we only want to have one sample (the first one we find) to reduce the number of samples returned to the frontend. 

If you want a more fine grained metrics graph you need to zoom into the time range.